src/shared_library.c

```diff
@@ -160,27 +160,49 @@ rcutils_ret_t
 rcutils_get_platform_library_name(
   const char * library_name,
   char * library_name_platform,
-  unsigned int buffer_size)
+  unsigned int buffer_size,
+  bool debug)
 {
   RCUTILS_CHECK_ARGUMENT_FOR_NULL(library_name, RCUTILS_RET_INVALID_ARGUMENT);
   RCUTILS_CHECK_ARGUMENT_FOR_NULL(library_name_platform, RCUTILS_RET_INVALID_ARGUMENT);
 
   int written = 0;
 
 #ifdef __linux__
-  if (buffer_size >= (strlen(library_name) + 7)) {
-    written = rcutils_snprintf(
-      library_name_platform, strlen(library_name) + 7, "lib%s.so", library_name);
+  if (debug) {
+    if (buffer_size >= (strlen(library_name) + 8)) {
+      written = rcutils_snprintf(
+        library_name_platform, strlen(library_name) + 8, "lib%sd.so", library_name);
+    }
+  } else {
+    if (buffer_size >= (strlen(library_name) + 7)) {
+      written = rcutils_snprintf(
+        library_name_platform, strlen(library_name) + 7, "lib%s.so", library_name);
+    }
   }
 #elif __APPLE__
-  if (buffer_size >= (strlen(library_name) + 10)) {
-    written = rcutils_snprintf(
-      library_name_platform, strlen(library_name) + 10, "lib%s.dylib", library_name);
+  if (debug) {
+    if (buffer_size >= (strlen(library_name) + 11)) {
+      written = rcutils_snprintf(
+        library_name_platform, strlen(library_name) + 11, "lib%sd.dylib", library_name);
+    }
+  } else {
+    if (buffer_size >= (strlen(library_name) + 10)) {
+      written = rcutils_snprintf(
+        library_name_platform, strlen(library_name) + 10, "lib%s.dylib", library_name);
+    }
   }
 #elif _WIN32
-  if (buffer_size >= (strlen(library_name) + 7)) {
```

Comment:
Just for the record: the `7` was incorrect before and is being fixed in this patch.

Reply:
maybe it's just not trivial for me to see, but would it make sense to set this hardcoded numbers to constants to give some semantics to it?